packages/components/popover/src/Popover.test.tsx

```diff
@@ -1,27 +1,318 @@
+import { useState } from 'react'
 import { describe, expect, it, vi } from 'vitest'
 
 import { Popover } from './Popover'
 
+const getSparkComponent = (componentName: string): Element | null => {
+  return document.querySelector(`[data-spark-component="${componentName}"]`)
+}
+
+mockResizeObserver()
+
 describe('Popover', () => {
-  it('should render', () => {
-    render(<Popover>Hello World!</Popover>)
+  describe('uncontrolled mode (stateful)', () => {
+    describe('opening', () => {
+      it('should open popover on click', async () => {
+        const user = userEvent.setup()
+
+        // Given a basic popover (closed by default)
+        render(
+          <Popover>
+            <Popover.Trigger asChild>
+              <button type="button">Click me</button>
+            </Popover.Trigger>
+            <Popover.Anchor>
+              <button type="button">Popover anchor</button>
+            </Popover.Anchor>
+            <Popover.Content>
+              <Popover.Arrow />
+              Popover content
+            </Popover.Content>
+          </Popover>
+        )
+
+        // ...by default, only trigger elements are rendered
+        expect(getSparkComponent('popover-anchor')).toBeInTheDocument()
+        expect(getSparkComponent('popover-trigger')).toBeInTheDocument()
+
+        // ...and content elements are not rendered
+        expect(getSparkComponent('popover-arrow')).toBeNull()
+        expect(getSparkComponent('popover-content')).toBeNull()
+        expect(screen.queryByText('Popover content')).not.toBeInTheDocument()
+
+        // When the user click on its associated trigger element
+        await user.click(screen.getByText('Click me'))
+
+        // Then the popover opens (content is visible)
+        expect(getSparkComponent('popover-arrow')).toBeInTheDocument()
+        expect(getSparkComponent('popover-content')).toBeInTheDocument()
+        expect(screen.getByText('Popover content')).toBeInTheDocument()
+      })
+
+      it('should force render using `forceMount` for SEO purposes', () => {
+        // Given a popover closed by default but using "forceMount" prop
+        render(
+          <Popover>
+            <Popover.Trigger asChild>
+              <button type="button">Click me</button>
+            </Popover.Trigger>
+            <Popover.Content forceMount>Popover content</Popover.Content>
+          </Popover>
+        )
+
+        // Then the content is rendered in the DOM
+        expect(screen.getByText('Popover content')).toBeInTheDocument()
+      })
+
+      describe('onOpenAutoFocus', () => {
+        it('should forward focus to first focusable child', () => {
+          // Given a onOpenAutoFocus event
+          const focusEvent = vi.fn()
+
+          // When an opened popover is rendered
+          render(
+            <Popover defaultOpen>
+              <Popover.Trigger asChild>
+                <button type="button">Click me</button>
+              </Popover.Trigger>
```

Comment:
Is the `type` needed here ?

Reply:
I remember that button should always have an explicit type :) https://dev.to/clairecodes/why-its-important-to-give-your-html-button-a-type-58k9